src/js/user.js

```diff
@@ -3,11 +3,36 @@
 var _ = require('lodash'),
     Gogs = require('gogs-client');
 
-function UserManager(auth, server) {
-
+function UserManager(auth, server) {   
+    
     var api = new Gogs(server + '/api/v1'),
         tokenStub = {name: 'btt-writer-desktop'};
 
+    const MAX_PAGE_SIZE = 50
+
+    const fetchRepoExhaustively = async function (uid, query, limit) {
```

Comment:
This is great.  More efficient and easier to maintain than the recursive strategy.  Good job.